modules_core/com.etendorx.auth/src/main/java/com/etendorx/auth/auth/AuthService.java

```diff
@@ -1,49 +1,78 @@
 import com.etendorx.auth.auth.hashing.PasswordHash;
 import com.etendorx.auth.auth.jwt.JwtRequest;
+import com.etendorx.auth.feign.ServiceClient;
 import com.etendorx.auth.feign.UserClient;
-import com.etendorx.auth.feign.UserModel;
+import com.etendorx.auth.feign.model.ServicesAccessModel;
+import com.etendorx.auth.feign.model.UserModel;
 import com.etendorx.utils.auth.key.JwtKeyUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.jsonwebtoken.Claims;
 import io.jsonwebtoken.impl.DefaultClaims;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.hateoas.CollectionModel;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
 import org.springframework.web.server.ResponseStatusException;
 
+import java.util.HashMap;
+import java.util.List;
+
 @Component
 @Slf4j
 public class AuthService {
 
   public static final String UNAUTHORIZED_MESSAGE = "Invalid username or password.";
   public static final String UNDEFINED_USERNAME_MESSAGE = "The username is not defined.";
   public static final String UNDEFINED_PASSWORD_MESSAGE = "The password is not defined.";
+  public static final String UNDEFINED_SERVICE_MESSAGE = "The service is not defined.";
+  public static final String UNDEFINED_SECRET_MESSAGE = "The secret is not defined.";
+  public static final String UNAUTHORIZED_SECRET_MESSAGE = "The secret is not match.";
+  public static final String UNAUTHORIZED_SERVICE_MESSAGE = "The service is not match.";
 
   public static final String PROJECTION = "auth";
+  public static final String HEADER_TOKEN = "X-TOKEN";
+  public static final String SUPER_USER_ORG_ID = "0";
+  public static final String EMPTY_SERVICE = "";
+  public static final String SUPER_USER_ID = "100";
 
   @Autowired
   private UserClient userClient;
 
+  @Autowired
+  private ServiceClient serviceClient;
+
   @Value("${token}")
   private String token;
 
   public void validateJwtRequest(JwtRequest jwtRequest) {
     log.debug("Running JWT request validation");
     String username = jwtRequest.getUsername();
     String password = jwtRequest.getPassword();
+    String service = jwtRequest.getService();
+    String secret = jwtRequest.getSecret();
 
     if (username == null || username.isEmpty()) {
       throw new ResponseStatusException(HttpStatus.BAD_REQUEST, UNDEFINED_USERNAME_MESSAGE);
     }
     if (password == null || password.isEmpty()) {
       throw new ResponseStatusException(HttpStatus.BAD_REQUEST, UNDEFINED_PASSWORD_MESSAGE);
     }
+    if (service == null || service.isEmpty()) {
```

Comment:
StringUtils.isEmpty(service) check if service is null or empty then you not needs check service == null before.

```
StringUtils.isEmpty(null)      = true
 StringUtils.isEmpty("")        = true
 StringUtils.isEmpty(" ")       = false
 StringUtils.isEmpty("bob")     = false
 StringUtils.isEmpty("  bob  ") = false
```